src/lib/y2network/sysconfig_routes_file.rb

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface"
+require "y2network/route"
+
+module Y2Network
+  # This class represents a file containing a set of routes
+  #
+  # @example Reading the default location, i.e., /etc/sysconfig/network/routes
+  #   file = Y2Network::SysconfigRoutesFile.new
+  #   file.load
+  #   file.routes.size #=> 2
+  #
+  # @example Reading routes for the interface eth1
+  #   file = Y2Network::SysconfigRoutesFile.new("/etc/sysconfig/network/ifroute-eth1")
+  #   file.load
+  #   file.routes.size #=> 1
+  class SysconfigRoutesFile
+    DEFAULT_ROUTES_FILE = "/etc/sysconfig/network/routes".freeze
+
+    # @return [Array<Route>] Routes
+    attr_reader :routes
+
+    # @param path [String] File path
+    def initialize(path = nil)
+      path ||= DEFAULT_ROUTES_FILE
```

Comment:
this looks strange, why not having DEFAULT_ROUTES_FILE as default parameter?

Reply:
In the beginning, I needed to distinguish between *no parameters* and the default value. But I've adjusted the code, and now it is not required anymore.